### DISCUSSION TITLE
How to deal with booleans and empty strings/lists?

### DISCUSSION THREAD

**Discussion by thiagomgd** [chhoumann/quickadd#556] [Q&A]
As title says: I'm trying to do a movie import from themoviedb, and I have a couple boolean prompts with questions for some extra data I want to add. I.e.: 
- watched: this sets `watched` to true, and also sets watchedDate to current date.
- with wife: if I watched with my wife or alone. If yes, adds her to `watchedWithPeople`. Otherwise, that's an empty string (since I guess we need to join arrays in a string.

The problem is that` false` and `""` leads quick add to prompt me for the values again...

Script Code:
```
const withWife = await QuickAdd.quickAddApi.yesNoPrompt("with wife?");
const watched = await QuickAdd.quickAddApi.yesNoPrompt("watched now?");

  QuickAdd.variables = {
    ...movie,
    withPeople: withWife ? "Lu" : "",
    watched: watched,
    watchedDates: watched ? new Date().toISOString() : "",
```

Template:
```
withPeople: {{VALUE:withPeople}}
watched: {{VALUE:watched}}
watchedDates: {{VALUE:watchedDates}}
```

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

This was a genuine bug back then, and it's fixed: since QuickAdd 2.0.0, only an *unset* (undefined) variable triggers a prompt. `""` and `false` are treated as intentional values, so your script works unchanged on current QuickAdd:

- `withPeople: ""` renders as empty, no re-prompt
- `watched: false` renders as `false` (which YAML reads as a boolean in frontmatter)

So `withPeople:` just comes out blank when you answer no, and `watched: false` lands as a proper boolean.

If you ever want prompt-side optionality too (skip a prompt and have the token resolve to nothing), there's `{{VALUE:x|optional}}` now: https://quickadd.obsidian.guide/docs/FormatSyntax#optional-fields

Closing as resolved.
